### ISSUE TITLE
mariadb fails to start with kernel 5.16.8 and btrfs

### ISSUE THREAD

**Issue by hax404** [NixOS/nixpkgs#160516]
### Describe the bug
On NixOS 21.11 Mariadb is broken with underlying btrfs and kernel 5.16.8 (`boot.kernelPackages = pkgs.linuxPackages_latest;`).

### Steps To Reproduce
1. Install new NixOS 21.11 on btrfs and with empty /var/lib/mysql
2. ```
   boot.kernelPackages = pkgs.linuxPackages_latest;
   services.mysql = {
     enable = true;
     package = pkgs.mariadb;
   };
   ```
3. `nixos-rebuild switch`
4. Start of mysql hangs and never comes up.

### Expected behavior
Mariadb starts normal.

### Screenshots
https://gist.github.com/hax404/0b4b285b16f7dcfc99db9eb18e8c7689

### Known Workarounds
 - Don't use linuxPackages_latest, instead use kernel 5.10
 - `services.mysql.settings.mysqld.innodb_use_native_aio = "OFF";`

### Additional context
Using git bisect, I found the commit https://github.com/NixOS/nixpkgs/commit/937cf30e7e7fff70b1ec6d1e8948d21aa8b24854 from PR https://github.com/NixOS/nixpkgs/pull/145117

Problems with newer kernels than 5.10 are known https://mariadb.com/kb/en/mariadb-1065-release-notes/#innodb

Also tested Mariadb 10.6.7 from PR https://github.com/NixOS/nixpkgs/pull/160384

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
@thoughtpolice @ajs124 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.16.8, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"nixos-21.11.336020.2128d0aa28e"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by ajs124** [comment]
As this seems to be an upstream issue in either mariadb, the kernel or both I opened an issue upstream [here](https://jira.mariadb.org/browse/MDEV-27900).

**Comment by grooverdan** [comment]
ref: [kernel bug report](https://lore.kernel.org/io-uring/CABVffEM0eEWho+206m470rtM0d9J8ue85TtR-A_oVTuGLWFicA@mail.gmail.com/T/#u).

**Comment by ajs124** [comment]
This should be fixed in the upcoming batch of mariadb releases.

**Comment by grooverdan** [comment]
And kernel too, so you were right about "or both". Thanks for the bug report.

**Comment by hax404** [comment]
As this is fixed on NixOS 22.05 with Kernel 5.18.12 and mariadb 10.6.8, I consider this Issue as closed.
